game/system/hid/sdl_util.cpp

```diff
@@ -0,0 +1,23 @@
+#include "sdl_util.h"
+
+#include "common/log/log.h"
+
+#include "third-party/SDL/include/SDL.h"
+
+namespace sdl_util {
+void log_error(const std::string& msg) {
+  std::string sdl_cause = SDL_GetError();
+  lg::error("SDL Error: {} - Cause: {}", msg, sdl_cause.empty() ? "n/a" : sdl_cause);
+}
+bool is_any_event_type(uint32_t event_type, std::vector<uint32_t> allowed_types) {
+  for (const auto& allowed_type : allowed_types) {
+    if (allowed_type == event_type) {
+      return true;
+    }
+  }
+  return false;
+}
+SDL_bool sdl_bool(const bool val) {
+  return val ? SDL_TRUE : SDL_FALSE;
```

Comment:
why does SDL need to do this.......

Reply:
from reading, it's because it's fundamentally a C library which doesn't have a proper boolean type

Reply:
yeah, this isn't an uncommon thing in C code... it just drives me crazy that people feel the need to make up stuff like `SDL_TRUE`, as if the concept of TRUE is somehow specific to SDL and not programming in general.

C has has `#include <stdbool.h>` to define `true`/`false` since c99, but I guess SDL has to be compatible with truly cursed platforms where `stdbool.h` doesn't exist, and it's not possible to define`true` as `1` for some crazy reason.